### DISCUSSION TITLE
How to enable or disable traffic in my map?

### DISCUSSION THREAD

**Discussion by anxheloo** [visgl/react-google-maps#388] [Q&A] [answered]
Hello guys, is there a way on how to enable or disable traffic layer on my map? I am not finding a way on how to do that. Thanks in advance

**Comment by usefulthink** [comment, marked as answer]
One way would be to create a component for that:

```tsx
const TrafficLayer = () => {
  const map = useMap();

  useEffect(() => {
    if(!map) return;

    const trafficLayer = new google.maps.TrafficLayer({map, autoRefresh: false});
    return () => { trafficLayer.setMap(null); };
  }, [map]);

  return <></>;
}
```

**Reply by anxheloo** [reply]
Is this component going to be rendered inside my <Map> ?  I am trying this approach:   const onLoad = useCallback((map) => {
    mapRef.current = map;
    
    if (!trafficLayerRef.current) {
      trafficLayerRef.current = new google.maps.TrafficLayer();
    } and than toggeling it using a Custom Button:   const toggleTrafficLayer = () => {
    console.log("inside toggle traffic layer");

    if (trafficLayerRef.current) {
      if (isTrafficVisible) {
        trafficLayerRef.current?.setMap(null);
      } else {
        trafficLayerRef.current?.setMap(mapRef.current);
      }
      setIsTrafficVisible(!isTrafficVisible);
    }
  }; but not working. 

  }, []);

**Reply by anxheloo** [reply]
Anyway, this is working, Thanks man
